bundle/edu.gemini.sp.vcs/src/main/scala/edu/gemini/sp/vcs/diff/MergePlan.scala

```diff
@@ -1,14 +1,112 @@
 package edu.gemini.sp.vcs.diff
 
-import edu.gemini.pot.sp.ISPProgram
+import edu.gemini.pot.sp._
+import edu.gemini.pot.sp.version.{VersionMap, NodeVersions}
+import edu.gemini.sp.vcs.diff.NodeDetail.Obs
+import edu.gemini.sp.vcs.diff.VcsFailure._
+import edu.gemini.spModel.rich.pot.sp._
 
 import scalaz._
+import Scalaz._
 
 /** Describes the modifications required for a local program to complete a
   * merge.
   */
 case class MergePlan(update: Tree[MergeNode], delete: Set[Missing]) {
 
+  /** "Encode" for serialization. The issue is that `scalaz.Tree` is not
+    * `Serializable` but we need to send `MergePlan`s over `trpc`.
+    */
+  def encode: MergePlan.Transport = {
+    def encodeTree(t: Tree[MergeNode]): MergePlan.TreeTransport =
+      MergePlan.TreeTransport(t.rootLabel, t.subForest.toList.map(encodeTree))
```

Comment:
The fact that `scalaz.Tree` is based on `Stream` is sort of unfortunate for this use case.  I don't think there is anywhere I can take advantage of it.  In particular here I need to build something that can be serialized and sent over `trpc`.  In our version of Scala at least `Stream` is not serializable.